src/build.h

```diff
@@ -0,0 +1,387 @@
+    toml::Value* top = &root;
+
+    auto widgets = top->setChild("widgets", toml::Array());
+    {
+        auto widget = widgets->push(toml::Table());
+        widget->setChild("type", "image");
+        widget->setChild("width", 1000);
+
+        auto topics = widget->setChild("topics", toml::Array());
+        {
+            auto topic = topics->push(toml::Table());
+            topic->setChild("topic", "some");
+            topic->setChild("count", 3);
+        }
+
+        {
+            auto topic = topics->push(toml::Table());
+            topic->setChild("topic", "something");
+            topic->setChild("count", 4);
+        }
+    }
+
+    {
+        auto widget = widgets->push(toml::Table());
+        widget->setChild("type", "foo");
+        widget->setChild("width", 2000);
+
+        auto topics = widget->setChild("topics", toml::Array());
+        {
+            auto topic = topics->push(toml::Table());
+            topic->setChild("topic", "bar");
+            topic->setChild("count", 5);
+        }
+    }
+
+    return root;
+}
+
+toml::Value build_array_table_02(void)
+{
+    toml::Value root((toml::Table()));
+    toml::Value* top = &root;
+
+    auto widgets = top->setChild("widgets", toml::Table());
+    widgets->setChild("type", "image");
+    widgets->setChild("width", 1000);
+
+    auto topics = widgets->setChild("topics", toml::Array());
+    {
+        auto topic = topics->push(toml::Table());
+        topic->setChild("topic", "some");
+        topic->setChild("count", 3);
+
+    }
+    {
+        auto topic = topics->push(toml::Table());
+        topic->setChild("topic", "something");
+        topic->setChild("count", 4);
+    }
+
+    return root;
+}
+
+toml::Value build_boolean_01(void)
+{
+    toml::Value root((toml::Table()));
+    toml::Value* top = &root;
+
+    top->setChild("bool1", true);
+    top->setChild("bool2", false);
+    top->setChild("true", true);
+    top->setChild("false", false);
+
+    return root;
+}
+
+toml::Value build_datetime_01(void)
+{
+    toml::Value root((toml::Table()));
+    toml::Value* top = &root;
+
+    {
+        std::tm t;
+        t.tm_sec = 0;
+        t.tm_min = 32;
+        t.tm_hour = 7;
+        t.tm_mday = 27;
+        t.tm_mon = 5 - 1;
+        t.tm_year = 1979 - 1900;
+        auto tp = std::chrono::system_clock::from_time_t(timegm(&t));
+
+        top->setChild("date1", tp);
+    }
+
+    {
+        std::tm t;
+        t.tm_sec = 0;
+        t.tm_min = 32;
+        t.tm_hour = 0;
+        t.tm_mday = 27;
+        t.tm_mon = 5 - 1;
+        t.tm_year = 1979 - 1900;
+        auto tp = std::chrono::system_clock::from_time_t(timegm(&t));
+
+        top->setChild("date2", tp);
+    }
+
+
+    {
+        std::tm t;
+        t.tm_sec = 0;
+        t.tm_min = 32;
+        t.tm_hour = 0;
+        t.tm_mday = 27;
+        t.tm_mon = 5 - 1;
+        t.tm_year = 1979 - 1900;
+        auto tp = std::chrono::system_clock::from_time_t(timegm(&t));
+
```

Comment:
I tried this, and that ended up with the equivalent writer_test failing (because the writer doesn't output microseconds)